File: MedArrival-Backend/src/main/java/com/cnesten/medarrivalbackend/DTO/PriceComponentDTO.java
package com.cnesten.medarrivalbackend.DTO;

import com.cnesten.medarrivalbackend.Models.Price.PriceComponentType;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class PriceComponentDTO {
    private Long id;
    private PriceComponentType componentType;
    private Float amount;
    private LocalDateTime effectiveFrom;
    private LocalDateTime effectiveTo;
    private ClientDTO client;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: MedArrival-Front/src/models/PriceComponentDTO.ts
import { BaseDTO } from "./BaseDTO";
import { ClientDTO } from "./ClientDTO";

export enum PriceComponentType {
    PURCHASE_PRICE = 'PURCHASE_PRICE',
    TRANSPORT = 'TRANSPORT',
    STORAGE = 'STORAGE',
    TRANSIT = 'TRANSIT',
    DUANE = 'DUANE',
    AMSSNUR = 'AMSSNUR'
}

export interface PriceComponentDTO extends BaseDTO {
    componentType: PriceComponentType;
    amount: number;
    effectiveFrom: Date;
    effectiveTo: Date | null;
    client: ClientDTO | null;
    usesDefaultPrice?: boolean;
}
